soupsieve/css_parser.py

```diff
@@ -212,6 +212,18 @@ def replace(m):
     return RE_CSS_ESC.sub(replace, string)
 
 
+class SelectorSyntaxError(SyntaxError):
+    """Syntax error in a CSS selector."""
+
+    def __init__(self, msg, pattern=None, index=None):
+        """Initialize."""
+
+        super(SelectorSyntaxError, self).__init__(msg)
+        self.text = pattern
+        self.lineno = 1
```

Comment:
Patterns can actually span multiple lines just like in real CSS: https://github.com/facelessuser/soupsieve/blob/master/soupsieve/css_parser.py#L966.

We probably need to calculate the line number.

Reply:
Something like this ought to work:
```python
if pattern is not None and index is not None:
    self.lineno = pattern.count('\n', 0, index) + 1
    self.offset = index - pattern.rfind('\n', 0, index)
```

but I wonder how the traceback formatter would cope with a multiline `text`.

The logic is sufficiently complicated that I'd like to have unit tests for it.  I'm unfamiliar with your test suite; do you think you could provide some hints?

Reply:
This isn't specific to a selector, so it can probably just go in `test_api.py`. While out test suite is using `pytest`, you can use normal `unittest` syntax. I'm currently using very little of the `pytest` features.  This doesn't require us to do anything complicated, so you don't need to test this in multiple parsers or anything, we just need to capture an assert.

```py3
    def test_selector_syntax_error_with_multilines(self):
        """Test that the selector syntax error properly accounts for multiple lines."""

        with self.assertRaises(SelectorSyntaxError) as e:
            sv.compile('[bad=attrib.bute]')
        # Do something with e to check it is as expected
```